README.md

```diff
@@ -5,757 +5,886 @@
-//      age: 0,
-//      address: {
-//        street: '',
-//        number: 0,
-//        city: 'Toronto'
-//      }
-//    }
+let blog = create(Blog, {
+  posts: [
+    { id: 1, title: "Hello World" },
+    { id: 2, title: "Most fascinating blog in the world" }
+  ]
+});
+
+blog.posts[0].state;
+//> Post<{ id: 1, title: 'Hello World' }>
+
+blog.posts[1].state;
+//> Post<{ id: 2, title: 'Most fascinating blog in the world' }>
 ```
 
-You can access transitions of composed states by using object property notation. Every time that you
-call a transition, you receive a new microstate and you start from the root of the original type.
+When you're working with parameterized types, the shape of the Microstate is determined by the value. In this case, `posts` is created with two items which will, in turn, create a Microstate with two items. Each item will be a Microstate of type `Post`. If you push another item onto the `posts` Microstate, it'll be treated as a `Post`.
 
 ```js
-create(Person)
-  .age.increment()
-  .address.city.set('San Francisco')
-  .address.street.set('Market St').state;
-// => {
-//      name: 'Taras',
-//      age: 1,
-//      address: {
-//        street: 'Market St',
-//        number: 0,
-//        city: 'San Francisco'
-//      }
-//    }
-```
-
-The objects can be of any complexity and can even support recursion.
+let blog2 = blog.posts.push({ id: 3, title: "It is only getter better" });
+
+blog2.posts[2].state;
+//> Post<{ id: 3, title: 'It is only getter better' }>
+```
+
+You can also create a parameterized object with `{Post}`. The difference is that the collection is treated as an object. This can be helpful when create normalized data stores.
 
 ```js
-class Person {
-  name = String;
-  father = Person;
+class Blog {
+  posts = { Post };
+}
+
+class Post {
+  id = Number;
+  title = String;
 }
 
-create(Person, { name: 'Stewie' })
-  .father.name.set('Peter')
-  .father.father.name.set('Mr Griffin')
-  .father.father.father.name.set('Mr Giffin Senior').state;
-// => { name: 'Stewie',
-//      father: {
-//        name: 'Peter',
-//        father: {
-//          name: 'Mr Griffin,
-//          father: {
-//            name: 'Mr Griffin Senior'
-//          }
-//        }
-//      }
-//    }
+let blog = create(Blog, {
+  posts: {
+    "1": { id: "1", title: "Hello World" },
+    "2": { id: 2, title: "Most fascinating blog in the world" }
+  }
+});
+
+blog.posts["0"].state;
+//> Post<{ id: 1, title: 'Hello World' }>
+
+blog.posts["1"].state;
+//> Post<{ id: 2, title: 'Most fascinating blog in the world' }>
+```
+
+Parameterized objects have `Object` type transitions which allow you to use `assign` and `put`.
+
+```js
+let blog2 = blog.posts.put("3", { id: 3, title: "It is only getter better" });
+
+blog2.posts["3"].state;
+//> Post<{ id: 3, title: 'It is only getter better' }>
 ```
 
-We currently do not support composition inside of arrays but we're looking into supporting it in the
-future.
+# State
```

Comment:
Do we need to rewrite this to de-emphasize state?

Reply:
Can you elaborate on "de-emphasize state"?

Reply:
@knownasilya Charles and spent the last 3 days re-thinking the role of getters in Microstates. There was previously a dichotomy between transitions and state. Getters were only on state and not on transitions because getters were used to automatically cache calculations on the state tree.

We had to roll back the caching of getters because extending JavaScript native classes with babel doesn't work. Between this and the fact that our structural sharing refactor changed the structure of microstates in a way that unlocked new capability, we're realizing that the state in it's previous conception might not be as useful.

State tree is immutable but when you pass it to components, you often need the transitions to make it useful. It's very rare that a component has readonly access to state. This got us thinking about how we could use getters better.

I now have a branch were we're intruding a feature that we talked about for a long time. It's a very small spike of what we're like to have ultimately, but it's enough to replace how we use getters.

This new feature is called querying. It allows to perform transitions but for the purpose of creating derivative state. Very similar to how use CPs in Ember the CP evaluates to another microstate that has transitions that you can invoke. The invoked transitions will be applied to the original objects that they reference in the original tree. 

This unlocks a lot of coolness that was not possible with previous state. With this in place, the state becomes less significant. The state property will still have the primitive value on it but its mostly to access value of the microstate.